### ISSUE TITLE
New delivery  - DHL

### ISSUE THREAD

**Issue by TarnJul** [allegro/allegro-api#9542]
Hi. We have started working with the new DHL delivery service. Manually we can send a parcel via Allegro and create a leis. But if we use the API, we get the error “UNKNOWN_ADDITIONAL_SERVICE”. What should I correct in the request?
Id Allegro 13046f50-4d68-11ef-a973-71f6de66839f

**Comment by PrzemyslawLukanowski** [comment]
Hi, instead of "ADDITIONAL_HANDLING" please try to provide "additionalHandling", the same way we return this information in the GET /shipment-management/delivery-services:

```
            "additionalServices": [
                {
                    "id": "additionalHandling",
                    "name": "Paczka niestandardowa",
                    "description": "Paczka o kształcie, który nie jest prostopadłościanem (np. opona, tuba) lub taka, która jest owinięta stretchem, folią lub innym materiałem utrudniającym przesuwanie."
                },
```

**Comment by TarnJul** [comment]
Hi. When delivery metod is DHL and dimention product is more than 120 x 60 x 60 cm I get this error DIMENSIONS_VALIDATION_ERROR. But the managers on the DHL side say that they work with large orders. Can you remove the limits for large orders from your side?

**Comment by Lukasz-Zurek** [comment]
We cannot remove the limits, because these are not on our side. Our logistic operator does have limits for every courier, also DHL, for specific methods. I believe that DHL works with large orders, but not in our Ship with Allegro service.

**Comment by TarnJul** [comment]
I see. Thanks.
I get another error DHL  "code": "PACKAGES_COUNT_EXCEEDED",  "message": "Maximum number of parcels is 1".
ID 9943b6e0-54aa-11ef-9546-d53b31e1912a
Is it problem on DHL side too? And why can users choose DHL delivery if they order more than 1 product? Why is there no checkout?

**Comment by Lukasz-Zurek** [comment]
You have provided more than one package in the "packages" section. It is not possible for DHL courier. The maximum number of packages included in one shipment: 10 (applies only to DPD and WE|DO).

So if you want to create more than one package, you have to create another shipment. It will be 1 shipment = 1 package in this situation.

**Comment by TarnJul** [comment]
Hi. Regarding the problem with large items. I contacted DHL support about large sizes. And they said that for large sizes there is an additional non-standard field in the “packages” section of the request. Can I add this field to the Allegro request?




> Мы не можем снять ограничения, потому что они не на нашей стороне. У нашего логистического оператора есть ограничения для каждого курьера, также DHL, для определенных методов. Я считаю, что DHL работает с большими заказами, но не в нашей службе Ship with Allegro.

**Comment by TarnJul** [comment]
Hi. Regarding the problem with large items. I contacted DHL support about large sizes. And they said that for large sizes there is an additional non-standard field in the “packages” section of the request. Can I add this field to the Allegro request?

> We cannot remove the limits, because these are not on our side. Our logistic operator does have limits for every courier, also DHL, for specific methods. I believe that DHL works with large orders, but not in our Ship with Allegro service.

**Comment by Lukasz-Zurek** [comment]
@TarnJul We do not support an additional field that would bypass the limit mentioned above in the Ship with Allegro service. But please precise what kind of field did DHL mention. What large size did they mean? What dimensions? I will consult this case with another department. Thank you for your message.

**Comment by TarnJul** [comment]
It it for shipment larger than 120/60/60

**Comment by Lukasz-Zurek** [comment]
@TarnJul Ok, I will ask another department if that kind of packages can be shipped using Ship with Allegro service.

**Comment by TarnJul** [comment]
Hi. Do you have some information?

**Comment by Lukasz-Zurek** [comment]
@TarnJul Yes, unfortunately we do not handle larger DHL packages. More information about max dimensions you can find [here](https://help.allegro.com/sell/en/a/allegro-dhl-courier-information-for-sellers-wwx8mjbBRu5#maximum-parcel-dimensions-and-weight).

**Comment by TarnJul** [comment]
Hello. Sent a parcel with dimensions 76.20*10.3*47.2, weight 5 kg. And then I received a message about additional charge for the parcel, because it is a non-standard parcel and one of the parameters exceeds 120 cm x 60 cm x 60 cm. 
Why the extra charge if everything is within the set dimensions?  "commandId": "4a9fffd0-212d-11f0-ad22-83bb00242ba1" "shipmentId": "aa533e71-df8d-4f4f-b22f-34a0a6fea08b"

![Image](https://github.com/user-attachments/assets/799536f6-7187-4ac6-a233-24fc805ff777)

**Comment by MaciejFrackowiak** [comment]
@TarnJul To explain additional charges, please contact [Allegro support](https://allegro.pl/pomoc/kontakt) directly.
